src/util/transaction.js

```diff
@@ -52,81 +61,207 @@ export const TRANSITIONS = [
+const STATE_ENQUIRY = 'enquiry';
+const STATE_PREAUTHORIZED = 'preauthorized';
+const STATE_DECLINED = 'declined';
+const STATE_ACCEPTED = 'accepted';
+const STATE_CANCELED = 'canceled';
+const STATE_DELIVERED = 'delivered';
+const STATE_REVIEWED = 'reviewed';
+const STATE_REVIEWED_BY_CUSTOMER = 'reviewed-by-customer';
+const STATE_REVIEWED_BY_PROVIDER = 'reviewed-by-provider';
 
-export const transitionsToRequested = [TRANSITION_REQUEST, TRANSITION_REQUEST_AFTER_ENQUIRY];
-export const txIsRequested = tx => transitionsToRequested.includes(txLastTransition(tx));
+/**
+ * Description of transaction process
+ *
+ * You should keep this in sync with transaction process defined Flex API
+ *
+ * Note: we don't use yet any state machine library,
+ * but this description format is following Xstate (FSM library)
+ */
 
-export const txIsAccepted = tx => txLastTransition(tx) === TRANSITION_ACCEPT;
+// Description format is following Xstate library (FSM)
+const stateDescription = {
+  id: 'preauth-nightly-booking',
 
-export const txIsDeclined = tx => txLastTransition(tx) === TRANSITION_DECLINE;
+  // This 'initial' state is a starting point for new transaction
+  initial: STATE_INITIAL,
 
-export const txIsExpired = tx => txLastTransition(tx) === TRANSITION_EXPIRE;
+  // States
+  states: {
+    [STATE_INITIAL]: {
+      on: {
+        [TRANSITION_ENQUIRE]: STATE_ENQUIRY,
+        [TRANSITION_REQUEST]: STATE_PREAUTHORIZED,
+      },
+    },
+    [STATE_ENQUIRY]: {
+      on: {
+        [TRANSITION_REQUEST_AFTER_ENQUIRY]: STATE_PREAUTHORIZED,
+      },
+    },
 
-export const txIsDeclinedOrExpired = tx => txIsDeclined(tx) || txIsExpired(tx);
+    [STATE_PREAUTHORIZED]: {
+      on: {
+        [TRANSITION_DECLINE]: STATE_DECLINED,
+        [TRANSITION_EXPIRE]: STATE_DECLINED,
+        [TRANSITION_ACCEPT]: STATE_ACCEPTED,
+      },
+    },
 
-export const txIsCanceled = tx => txLastTransition(tx) === TRANSITION_CANCEL;
+    [STATE_DECLINED]: {},
+    [STATE_ACCEPTED]: {
+      on: {
+        [TRANSITION_CANCEL]: STATE_CANCELED,
+        [TRANSITION_COMPLETE]: STATE_DELIVERED,
+      },
+    },
+
+    [STATE_CANCELED]: {},
+    [STATE_DELIVERED]: {
+      on: {
+        [TRANSITION_EXPIRE_REVIEW_PERIOD]: STATE_REVIEWED,
+        [TRANSITION_REVIEW_1_BY_CUSTOMER]: STATE_REVIEWED_BY_CUSTOMER,
+        [TRANSITION_REVIEW_1_BY_PROVIDER]: STATE_REVIEWED_BY_PROVIDER,
+      },
+    },
+
+    [STATE_REVIEWED_BY_CUSTOMER]: {
+      on: {
+        [TRANSITION_REVIEW_2_BY_PROVIDER]: STATE_REVIEWED,
+        [TRANSITION_EXPIRE_PROVIDER_REVIEW_PERIOD]: STATE_REVIEWED,
+      },
+    },
+    [STATE_REVIEWED_BY_PROVIDER]: {
+      on: {
+        [TRANSITION_REVIEW_2_BY_CUSTOMER]: STATE_REVIEWED,
+        [TRANSITION_EXPIRE_CUSTOMER_REVIEW_PERIOD]: STATE_REVIEWED,
+      },
+    },
+    [STATE_REVIEWED]: { type: 'final' },
+  },
+};
+
+// Note: currently we assume that state config doesn't contain nested states.
+const statesFromConfig = config => config.states || {};
```

Comment:
I first thought `config` meant config.js file so could we use `stateConfig` here too?